Udp fixes

fabbione opened the pull request.

fabbione committed:
[transports] use SO_REUSEADDR only for sctp

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

wferi commented on libknet/tests/test-common.c:
```diff
@@ -445,18 +445,27 @@ int knet_handle_stop(knet_handle_t knet_h)
+		}
+		if (port > 65536) {
+			port = port & 0xFFFF;
+		}
```
This can leave you with any 0 <= port <= 65536, which is wrong on both ends. How about this?
```suggestion
		/* Use the pid if we can. but makes sure its in a sensible range */
		port = (getpid() + offset) % (65536-1024) + 1024;
```

fabbione replied:
make sense to me, i´ll wait for @chrissie-c to add her review and I´ll change it.

wferi commented on libknet/tests/test-common.c:
```diff
@@ -445,18 +445,27 @@ int knet_handle_stop(knet_handle_t knet_h)
-static int _make_local_sockaddr(struct sockaddr_storage *lo, uint16_t offset, int family)
+static int _make_local_sockaddr(struct sockaddr_storage *lo, int offset, int family)
 {
 	uint32_t port;
```
Not important, just may make things more obvious/explicit:
```suggestion
	in_port_t port;
```

fabbione replied:
same as above

chrissie-c reviewed (dismissed):
ACK with Feri's modifications.

fabbione committed:
[tests] rework test suite link port allocation

Logic is to try to configure a link with port X and if it fails, try the next
port. This avoids port collisions between services and knet test suite.

Please note that the implementation in test-common.c is NOT super clean.
There is still some redundant code in there that is left on purpose.

There is another branch, not yet merged, that implements functional testing
framework that does heavy use of those functions.

We will clean test-common.c as we port the functional testing branch and make
it ready for merging.

For now, this is good enough to have a more stable test suite.

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

chrissie-c reviewed (approved):
re-ACK

fabbione commented:
retest this

fabbione commented:
retest this please
